### ISSUE TITLE
Http client pool causes TypeError on bad redirect

### ISSUE THREAD

**Issue by johanobergman** [laravel/framework#44014]
- Laravel Version: 9.24.0
- PHP Version: 8.1.9
- Database Driver & Version:

### Description:
Making an http request using the Laravel Http client throws `GuzzleHttp\Psr7\Exception\MalformedUriException` if the called endpoint has a bad redirect header.

That's ok and can be caught in case of a single http call, but when using the Http Pool feature one bad redirect crashes the entire pool of requests with a PHP error: `Illuminate\Http\Client\PendingRequest::Illuminate\Http\Client\{closure}(): Argument #1 ($e) must be of type GuzzleHttp\Exception\TransferException, GuzzleHttp\Psr7\Exception\MalformedUriException given`.

The relevant line is here: https://github.com/laravel/framework/blob/9.x/src/Illuminate/Http/Client/PendingRequest.php#L883

Laravel type hints `TransferException`, which a `MalformedUriException` is not, and causes a PHP TypeError.

Could a looser type hint that covers all possible Guzzle exceptions be a solution? I don't want to throw away all responses just because one response is bad.

### Steps To Reproduce:
Using a site named `example-site` in Laravel Valet.

The PHP built in server will timeout instead of returning the bad redirect header on /bad-redirect.

routes/web.php

```php
use Illuminate\Support\Facades\Http;

// This endpoint would be somewhere on the internet and not in my own codebase, but for the sake of testing
// this will do.
Route::get('/bad-redirect', function() {
    return redirect('//');
});

Route::get('/test', function () {
    // This triggers MalformedUriException but can be caught easily.
    // $response = Http::get('http://example-site.test/bad-redirect');

    // This triggers a PHP TypeError due to the MalformedUriException not being a TransferException.
    $responses = Http::pool(fn($pool) => $pool->get('http://example-site.test/bad-redirect'));

    return $responses[0]->ok();
});
```

**Comment by driesvints** [comment]
I don't get the same error. I get the following:

<img width="1467" alt="Screenshot 2022-09-06 at 08 38 39" src="https://user-images.githubusercontent.com/594614/188564093-50aca02a-f6f4-4f5e-9d62-59c600693f8a.png">

**Comment by driesvints** [comment]
Hey there,

Can you first please try one of the support channels below? If you can actually identify this as a bug, feel free to open up a new issue with a link to the original one and we'll gladly help you out.

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

Thanks!

**Comment by johanobergman** [comment]
I don't think it should return a ConnectException either.

But what if you try this url: https://www.smol.se/arrangemang2/Tillstartsidan/

The live one I had trouble with.

**Comment by johanobergman** [comment]
curl --head https://www.smol.se/arrangemang2/Tillstartsidan/ returns Location: //

**Comment by johanobergman** [comment]
Ran it again with a completely fresh Laravel install. Took a few seconds but then I got this. Maybe your web server behaves differently than mine?

Still thats just to simulate the error - once you hit a real server that behaves that way it will throw a TypeError. Try the url I provided above, that should give you the same response I get.

<img width="1017" alt="Skärmavbild 2022-09-06 kl  08 49 59" src="https://user-images.githubusercontent.com/6485774/188566065-26f721a5-5825-49c2-9590-42586b50bc3e.png">
